modules/apps/static/portal-configuration/portal-configuration-persistence/src/main/java/com/liferay/portal/configuration/persistence/ConfigurationPersistenceManager.java

```diff
@@ -320,10 +322,19 @@ protected void doStore(
 
 		ConfigurationModelListener configurationModelListener = null;
 
-		if (hasPid(pid)) {
+		if (!pid.endsWith("factory") &&
+			(dictionary.get("_felix_.cm.newConfiguration") == null)) {
```

Comment:
@rotty3000 can you give Drew any clue on how to perform this check in a more elegant way?